### PR TITLE
feat: separate envoy lds configs for self-hosting and supabase use-cases

### PR DESCRIPTION
**Pull request by hf**
Separates Envoy's `lds.yaml` file for two different use cases: self-hosting and for Supabase internal use.

For self-hosting:

- File remains to be called `lds.yaml`
- Includes only self-hosing concepts such as fixed API keys
- Control which `lds.yaml` file is used with the `envoy_lds` Docker build arg

For Supabase:

- File is called `lds.supabase.yaml`
- Is replaced into `/etc/envoy/lds.yaml` with an Ansible task part of `ansible/tasks/setup-supabase-internal.yml`
- For Fly, replaced using the `envoy_lds` Docker build arg
- In follow up PR it will contain an origin protection key which is not something available in the Docker image for self-hosting
- Future: once origin protection keys are rolled out and new API keys are in place, Envoy will no longer check any API keys

### CODE REVIEW

File: ansible/files/envoy_config/lds.supabase.yaml [hunk 1]
```diff
@@ -0,0 +1,436 @@
+resources:
```
Comment (hf) [inline ansible/files/envoy_config/lds.supabase.yaml:1]: A verbatim copy of `lds.yaml`

**Comment by hf** [comment]
Will bump versions after this is approved.

File: ansible/tasks/internal/setup-envoy.yml
Comment (hf) [inline ansible/tasks/internal/setup-envoy.yml:1]: Remove, not needed.

**Review by pcnc — APPROVED** [review]

File: ansible/vars.yml [hunk 2]
```diff
@@ -11,8 +11,8 @@ postgres_major:
 
 # Full version strings for each major version
 postgres_release:
-  postgres15: "15.8.1.010"
-  postgres16: "16.3.1.016"
+  postgres15: "15.8.1.011"
+  postgres16: "16.3.1.017"
```
Comment (hf) [inline ansible/vars.yml:16]: @samrose @supabase/postgres OK to merge with these?

**Review by samrose — APPROVED** [review]
Ok to merge

**Pull request merged**